ppx/melange_ppx.ml

```diff
@@ -954,6 +951,28 @@ module Derivers = struct
     in
     Deriving.add ~str_type_decl ~sig_type_decl "abstract"
 
+  let record_constructor =
+    let args () = Deriving.Args.empty in
+    let str_type_decl =
+      Deriving.Generator.V2.make (args ()) (fun ~ctxt:_ (rf, tdcls) ->
+          Ast_derive_abstract.handleCstrTdclsInStr rf tdcls)
+    and sig_type_decl =
+      Deriving.Generator.V2.make (args ()) (fun ~ctxt:_ (rf, tdcls) ->
+          Ast_derive_abstract.handleCstrTdclsInSig rf tdcls)
+    in
+    Deriving.add ~str_type_decl ~sig_type_decl "make_opt_keys"
```

Comment:
is "make" necessary, or could it be just `opt_keys`? or `opt_js_keys` as you [mentioned](https://github.com/melange-re/melange/pull/979#issuecomment-1856183928) including "js" somewhere.

Reply:
yeah, I'm open to changing the name; I'd prefer keeping something that mentions it makes a JS object, e.g. `jsConstructor` or `jsObject`

Reply:
I was thinking about the JS [hasOwnProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty) method. It seems a property is the mix of [a key and a value](https://developer.mozilla.org/en-US/docs/Glossary/Property/JavaScript), so maybe the deriver name should include it.

It'd be nice to define it like "constructor that skip properties in the JS object if the field is `None`", maybe `optJsProperties`, `makeOptProperties`? or `skipPropertiesConstructor`?

Reply:
> I'd prefer keeping something that mentions it makes a JS object,

I am not sure this is very informative. All records compile now to JS objects so a name related to that wouldn't be very meaningful imo.

Reply:
ok, I called it `jsProperties` to unblock the PR, but we can brainstorm a better name later on.